x/qgb/keeper/msg_server.go

```diff
@@ -45,8 +47,56 @@ func (k msgServer) ValsetConfirm(c context.Context, msg *types.MsgValsetConfirm)
 }
 
 // DataCommitmentConfirm handles MsgDataCommitmentConfirm
-func (k msgServer) DataCommitmentConfirm(context.Context, *types.MsgDataCommitmentConfirm) (*types.MsgDataCommitmentConfirmResponse, error) {
-	// TODO
+func (k msgServer) DataCommitmentConfirm(c context.Context, msg *types.MsgDataCommitmentConfirm) (*types.MsgDataCommitmentConfirmResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	sigBytes, err := hex.DecodeString(msg.Signature)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrInvalid, "signature decoding")
+	}
+
+	ethAddress, err := types.NewEthAddress(msg.EthAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrInvalid, "invalid eth address")
+	}
+
+	// TODO uncomment when we add slashing module
+	//_, err = sdk.AccAddressFromBech32(msg.ValidatorAddress)
+	//if err != nil {
+	//	return nil, sdkerrors.Wrap(types.ErrInvalid, "validator address invalid")
+	//}
+	//validator, found := GetAccountValidator(ctx, validatorAddress)
+	//if !found {
+	//	return nil, sdkerrors.Wrap(types.ErrUnknown, "validator")
+	//}
+	//if err := sdk.VerifyAddressFormat(validator.GetOperator()); err != nil {
+	//	return nil, sdkerrors.Wrapf(err, "discovered invalid validator address for validator %v", validatorAddress)
+	//}
+
+	// TODO get eth address from store
+	//ethAddressFromStore, found := k.GetEthAddressByValidator(ctx, validator.GetOperator())
+	//if !found {
+	//	return nil, sdkerrors.Wrap(types.ErrEmpty, "no eth address set for validator")
+	//}
+	//
+	//if *ethAddressFromStore != *submittedEthAddress {
+	//	return nil, sdkerrors.Wrap(types.ErrInvalid, "submitted eth address does not match delegate eth address")
+	//}
+
+	err = types.ValidateEthereumSignature([]byte(msg.Commitment), sigBytes, *ethAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrInvalid, fmt.Sprintf("signature verification failed expected sig by %s with checkpoint %s found %s", ethAddress, msg.Commitment, msg.Signature))
+	}
+
+	// TODO set data commitment confirm when we start persisting state
```

Comment:
I don't think it matters exactly when, but out of curiosity, will this be included in this PR or another?

Reply:
In this PR. This PR is still WIP. I just push to have a better vision on what is happening and how it looks when reviewing :D